src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestHelpers/BaseRunner.cs

```diff
@@ -14,6 +14,65 @@
 
 namespace Microsoft.PowerFx.Core.Tests
 {
+    /// <summary>
+    /// Result of a call to <see cref="BaseRunner.RunAsyncInternal(string, string)"/>.
+    /// - Compilation error - set Errors
+    /// - Runtime error - set Value
+    ///
+    /// If the eval used unsupported behavior, set UnsupportedReason to a message.
+    /// This may cause th etest to get skipped rather than fail. 
+    /// </summary>
+    public class RunResult
```

Comment:
>RunResult [](http://example.com/codeflow?start=17&length=9)

Before, we would throw in some cases, return values in others.  This makes it more consistent - which is important since we want to allow a UnsupportedReason in any scenario.